### DISCUSSION TITLE
Get address of Long Array?

### DISCUSSION THREAD

**Discussion by GitRayman** [micropython/micropython#15899] [Using MicroPython] [answered]
Forgive me if this is obvious, having trouble figuring this out...

Have an array like this:
'self.pins = array.array('L',[self.basepin, self.rst, self.rd, self.wr, self.dc, self.cs])'

Would like to get the physical address of the array in the Heap, so can pass to a driver....

I've tried several things with uctypes, but can't get it to work...
Things are simple with bytearray, but can't find a solution here..
Any ideas?

Looks like could create a structure with uctypes instead and get the address that way.
But, is there an easier way?  Feels like there should be...

**Comment by mendenm** [comment, marked as answer]
```uctypes.addressof(obj)```
from 
(https://docs.micropython.org/en/latest/library/uctypes.html#uctypes.addressof)
should work, even if you didn't create the object with uctypes, as long it has the buffer protocol (which your array does).

**Reply by GitRayman** [reply]
Thanks!  That works.  Think tried this before, but must have done it wrong...
